core/src/main/java/hudson/model/Actionable.java

```diff
@@ -116,7 +130,11 @@
      * @see #getAction(Class)
      */
     public <T extends Action> List<T> getActions(Class<T> type) {
-        return Util.filter(getAllActions(), type);
+        List<T> _actions = Util.filter(getActions(), type);
+        for (TransientActionFactory<?> taf : TransientActionFactory.factoriesFor(getClass(), type)) {
+            _actions.addAll(Util.filter(createFor(taf), type));
+        }
+        return Collections.unmodifiableList(_actions);
```

Comment:
I agree with Jesse. Modification of the filtered list is not a good idea in any case.
We either expose the internal representation or send changes to /dev/null

BTW, it would be great to Javadoc the fact that the list should not be modified